src/compas_fab/backends/pybullet/client.py

```diff
@@ -0,0 +1,410 @@
+        body_id = self.body_id_or_default(body_id)
+        return list(range(self.get_num_joints(body_id)))
+
+    def get_joint_name(self, joint_id, body_id=None):
+        body_id = self.body_id_or_default(body_id)
+        return self.get_joint_info(joint_id, body_id).jointName
+
+    def get_link_name(self, link_id, body_id=None):
+        body_id = self.body_id_or_default(body_id)
+        if link_id == BASE_LINK_ID:
+            return self.get_base_name(body_id)
+        return self.get_joint_info(link_id, body_id).linkName.decode('UTF-8')
+
+    def get_link_frame(self, link_id, body_id=None):
+        body_id = self.body_id_or_default(body_id)
+        if link_id == BASE_LINK_ID:
+            return self.get_base_frame(body_id)
+        link_state = self.get_link_state(link_id, body_id)
+        pose = (link_state.worldLinkFramePosition, link_state.worldLinkFrameOrientation)
+        return frame_from_pose(pose)
+
+    # =======================================
+    def set_base_frame(self, frame, body_id=None):
+        # !!! This could lead to a discrepancy between the self._robot and what's going on in pybullet.
+        # !!! RCF would not match
+        body_id = self.body_id_or_default(body_id)
+        point, quaternion = pose_from_frame(frame)
+        pybullet.resetBasePositionAndOrientation(body_id, point, quaternion, physicsClientId=self.client_id)
+
+    def set_joint_position(self, joint_id, value, body_id=None):
+        body_id = self.body_id_or_default(body_id)
+        pybullet.resetJointState(body_id, joint_id, value, targetVelocity=0, physicsClientId=self.client_id)
+        # self.create_collision_map() ???
+
+    def set_joint_positions(self, joints, values, body_id=None):
+        body_id = self.body_id_or_default(body_id)
+        if len(joints) != len(values):
+            raise Exception('Joints and values must have the same length.')
+        for joint, value in zip(joints, values):
+            self.set_joint_position(joint, value, body_id)
+
+    # =======================================
+    def convert_mesh_to_body(self, mesh, frame, _name=None):
+        """Convert compas mesh and its frame to a pybullet body
+
+        Parameters
+        ----------
+        mesh : `compas.datastructures.Mesh`
+        frame : :class:`compas.geometry.Frame`
+        _name : :obj:`str`, optional
+            Name of the mesh for tagging in pybullet's GUI
+
+        Returns
+        -------
+        :obj:`int`
+        """
+        temp_dir = tempfile.mkdtemp()
+        tmp_obj_path = os.path.join(temp_dir, 'temp.obj')
+        mesh.to_obj(tmp_obj_path)
+        pyb_body_id = self.body_from_obj(tmp_obj_path)
+        self.set_base_frame(frame, pyb_body_id)
+        # !!! The following lines are apparently for visual debugging purposes
+        # To be deleted or rewritten later.
+        # if name:
+        #     pybullet_planning.add_body_name(pyb_body, name)
+        os.remove(tmp_obj_path)
```

Comment:
wrap this temp dir stuff into a `try..finally` block so that we make sure we remove the temp dir/file even if the middle part crashes.